functions/common.py

```diff
@@ -1,26 +1,108 @@
 import exceptions.exceptions as exceptions
 
 
-def factorial(num: int) -> int:
-    if not isinstance(num, int):
-        # Input should be an integer
-        raise exceptions.InputError(num, "Error: non-integer factorial")
-    elif num < 0:
-        # Input should be non-negative
-        raise exceptions.InputError(num, "Error: negative factorial")
-    elif num == 0:
-        return 1
+def factorial(argument: int) -> int:
+    """Returns the product of all non-negative integers less than or equal to some non-negative integer.
+
+    Args:
+        argument (int): Non-negative integer used as input to factorial function
+
+    Returns:
+        int: Product of all non-negative integers less than or equal to 'argument'
+        
+    Raises:
+        InputError: If 'argument' is not a non-negative integer
+    """
+    # Ensure that input is an integer.
+    if int(argument) != argument:
+        raise exceptions.InputError(argument, "Input to factorial should be a non-negative integer.")
+    argument = int(argument)
+    # Ensure that input is non-negative.
+    if argument < 0:
+        raise exceptions.InputError(argument, "Input to factorial should be a non-negative integer.")
+        
+    # Calculate factorial using repeated multiplication.
     result = 1
-    for i in range(1, num + 1):
+    for i in range(1, argument + 1):
         result *= i
     return result
 
 
-def mod(dividend: float, divisor: float) -> float:
-    # TODO not sure on the typing for this,
-    # and should check if can be optimized
-    while dividend > divisor:
-        dividend -= divisor
-    while dividend < divisor:
-        dividend += divisor
-    return dividend
+def is_even(argument: int) -> bool:
+    """Returns true is input is even and false otherwise.
+
+    Args:
+        argument (int): Value to be checked
+
+    Returns:
+        bool: True if 'argument' is even and false otherwise
+    """
+    return not (argument & 1)
```

Comment:
May want to do type checking here and raise exception when argument is not an int. 
This would also apply for is_odd

Reply:
I think I left it out to avoid spending cycles on it but you're right it probably should be there. I'll add it.